proofs/tree.go

```diff
@@ -747,14 +753,24 @@ func newMessageFlattener(message, messageSalts proto.Message, saltsLengthSuffix
 	return &f
 }
 
+func ensurePrefixTrailingDot(prefix string) string {
+	out := prefix
+	if out != "" {
+		if isDot := out[len(out)-1]; isDot != '.' {
+			out = out + "."
+		}
+	}
```

Comment:
```
	if !strings.HasSuffix(prefix, ".") {
		prefix = prefix + "."
	}
```

This should handle all the edge cases IMO. What do you think?

Reply:
ok, I just need to add only when prefix is not empty. Pushing now.